pkg/util/condition_test.go

```diff
@@ -0,0 +1,154 @@
+package util
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kubevirt-image-service/pkg/apis/hypercloud/v1alpha1"
+	"time"
+)
+
+var _ = Describe("GetConditionByType", func() {
+	Context("if conditions has no matching conditionType", func() {
+		found, _ := GetConditionByType([]v1alpha1.Condition{
+			{
+				Type:               "type1",
+				Status:             corev1.ConditionTrue,
+				ObservedGeneration: 32,
```

Comment:
Why do we need ObservedGeneration in the test code? It seems that the value is not used in the test code.

Reply:
As far as I understand it's an optional value suggested by [the proposal](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1623-standardize-conditions#noteworthy-choices) to identify outdated condition compare to the .metadata.generation value. Also `ObservedGeneration` can be used to distinguish values between intentionally setted zero ones versus unsetted ones.

Reply:
We need to make sure that other values of condition are not changed when calling the function.